python/virgo/mpi/parallel_hdf5.py

```diff
@@ -570,6 +582,18 @@ def read(self, datasets, group=None, return_file_nr=None, read_attributes=False,
             for name in file_nr:
                 file_nr[name] = virgo.mpi.util.replace_none_with_zero_size(file_nr[name], self.comm)
 
+        # Ensure all ranks have the same dtype
```

Comment:
I wasn't expecting the dtype check to happen twice for independent reads but I guess it makes sense. The check in _read_independent is between files read by one rank but we could still have different dtypes between ranks.